File: backend/app/schemas/nfl.py
"""NFL week, game, and confidence-pick API schemas."""

import uuid
from datetime import datetime

from app.schemas.base import CamelModel


class WeekRead(CamelModel):
    id: uuid.UUID
    season: int
    week_number: int
    start_date: datetime
    end_date: datetime
    status: str
    locks_at: datetime | None = None
    is_locked: bool = False


class GameRead(CamelModel):
    id: uuid.UUID
    away_team: str
    home_team: str
    kickoff: datetime
    status: str
    venue_name: str | None = None
    venue_location: str | None = None
    spread_team: str | None = None
    spread: float | None = None
    away_score: int | None = None
    home_score: int | None = None
    winning_team: str | None = None
    is_tie: bool


class PickRead(CamelModel):
    id: uuid.UUID
    game_id: uuid.UUID
    team: str
    confidence: int
    submitted_at: datetime


class HistoricalPickRead(CamelModel):
    id: uuid.UUID
    game_id: uuid.UUID
    away_team: str
    home_team: str
    kickoff: datetime
    status: str
    team: str
    confidence: int
    submitted_at: datetime
    winning_team: str | None = None
    is_tie: bool
    points_earned: int | None = None
    outcome: str


class HistoricalWeekRead(CamelModel):
    week_number: int
    picks: list[HistoricalPickRead]


class PickHistoryRead(CamelModel):
    season: int
    weeks: list[HistoricalWeekRead]


class PickCreate(CamelModel):
    game_id: uuid.UUID
    team: str
    confidence: int


class PicksCreateRequest(CamelModel):
    week: int
    picks: list[PickCreate]

File: frontend/src/types/nfl.ts
export interface NflWeek {
  id: string
  season: number
  weekNumber: number
  startDate: string
  endDate: string
  status: string
  locksAt?: string | null
  isLocked?: boolean
}

export interface NflGame {
  id: string
  awayTeam: string
  homeTeam: string
  kickoff: string
  status: string
  venueName: string | null
  venueLocation: string | null
  spreadTeam: string | null
  spread: number | null
  awayScore: number | null
  homeScore: number | null
  winningTeam: string | null
  isTie: boolean
}

export interface NflPick {
  id: string
  gameId: string
  team: string
  confidence: number
  submittedAt: string
}

export type PickOutcome = 'correct' | 'incorrect' | 'unscored'

export interface HistoricalPick {
  id: string
  gameId: string
  awayTeam: string
  homeTeam: string
  kickoff: string
  status: string
  team: string
  confidence: number
  submittedAt: string
  winningTeam: string | null
  isTie: boolean
  pointsEarned: number | null
  outcome: PickOutcome
}

export interface HistoricalWeek {
  weekNumber: number
  picks: HistoricalPick[]
}

export interface PickHistory {
  season: number
  weeks: HistoricalWeek[]
}

export interface PickInput {
  gameId: string
  team: string
  confidence: number
}

export interface SavePicksInput {
  week: number
  picks: PickInput[]
}
